vm/systemSmartContracts/esdt.go

```diff
@@ -162,15 +171,20 @@ func (e *esdt) issueProtected(args *vmcommon.ContractCallInput) vmcommon.ReturnC
 		e.eei.AddReturnMessage("invalid owner address length")
 		return vmcommon.FunctionWrongSignature
 	}
-	if args.CallValue.Cmp(e.baseIssuingCost) != 0 {
-		e.eei.AddReturnMessage("callValue not equals with baseIssuingCost")
-		return vmcommon.OutOfFunds
-	}
 	err := e.eei.UseGas(e.gasCost.MetaChainSystemSCsCost.ESDTIssue)
 	if err != nil {
-		e.eei.AddReturnMessage("not enough gas")
+		e.eei.AddReturnMessage(err.Error())
 		return vmcommon.OutOfGas
 	}
+	esdtConfig, err := e.getESDTConfig()
+	if err != nil {
+		e.eei.AddReturnMessage(err.Error())
+		return vmcommon.UserError
+	}
+	if args.CallValue.Cmp(esdtConfig.BaseIssuingCost) != 0 {
+		e.eei.AddReturnMessage("callValue not equals with baseIssuingCost")
+		return vmcommon.OutOfFunds
```

Comment:
is this an OutOfFunds error or an userError?

Reply:
yes